File: app/src/main/java/code/name/monkey/retromusic/mvp/contract/GenreDetailsContract.java
package code.name.monkey.retromusic.mvp.contract;

import code.name.monkey.retromusic.model.Song;
import code.name.monkey.retromusic.mvp.BasePresenter;
import code.name.monkey.retromusic.mvp.BaseView;

import java.util.ArrayList;

/**
 * @author Hemanth S (h4h13).
 */

public interface GenreDetailsContract {
    interface GenreDetailsView extends BaseView<ArrayList<Song>> {
    }

    interface Presenter extends BasePresenter<GenreDetailsView> {
        void loadGenre(int genreId);
    }
}

File: app/src/main/java/code/name/monkey/retromusic/mvp/contract/GenreDetailsContract.kt
package code.name.monkey.retromusic.mvp.contract

import code.name.monkey.retromusic.model.Song
import code.name.monkey.retromusic.mvp.BasePresenter
import code.name.monkey.retromusic.mvp.BaseView

import java.util.ArrayList

/**
 * @author Hemanth S (h4h13).
 */

interface GenreDetailsContract {
    interface GenreDetailsView : BaseView<ArrayList<Song>>

    interface Presenter : BasePresenter<GenreDetailsView> {
        fun loadGenre(genreId: Int)
    }
}
